core/lib/Thelia/Core/Stack/ParamInitMiddleware.php

```diff
@@ -134,7 +134,14 @@ protected function defineCurrency(TheliaRequest $request)
     protected function detectLang(TheliaRequest $request)
     {
         // first priority => lang parameter present in request (get or post)
-        if ($request->query->has("lang")) {
+        $requestedLangCodeOrLocale = $request->query->has("lang");
+
+        // add a fallback on locale parameter
+        if (null === $requestedLangCodeOrLocale) {
```

Comment:
I think you have to compare with `false`. The `has` method will return `true` or `false`